src/bench/connectblock.cpp

```diff
@@ -0,0 +1,145 @@
+ * - For Taproot outputs: Creates simple key path spendable outputs
+ * - For non-Taproot outputs: Creates P2WPKH (native SegWit v0) outputs
+ * - All outputs have value of 1 BTC
+ */
+std::pair<std::vector<CKey>, std::vector<CTxOut>> CreateKeysAndOutputs(const CKey& coinbaseKey, size_t num_taproot, size_t num_nontaproot)
+{
+    std::vector<CKey> keys{coinbaseKey};
+    keys.reserve(num_taproot + num_nontaproot + 1);
+
+    std::vector<CTxOut> outputs;
+    outputs.reserve(num_taproot + num_nontaproot);
+
+    for (size_t i{0}; i < num_nontaproot; i++) {
+        const CKey key{GenerateRandomKey()};
+        keys.emplace_back(key);
+        outputs.emplace_back(COIN, GetScriptForDestination(WitnessV0KeyHash{key.GetPubKey()}));
+    }
+
+    for (size_t i{0}; i < num_taproot; i++) {
+        CKey key{GenerateRandomKey()};
+        keys.emplace_back(key);
+        outputs.emplace_back(COIN, GetScriptForDestination(WitnessV1Taproot{XOnlyPubKey(key.GetPubKey())}));
+    }
+
+    return {keys, outputs};
+}
+
+void BenchmarkConnectBlock(benchmark::Bench& bench, std::vector<CKey>& keys, std::vector<CTxOut>& outputs, TestChain100Setup& test_setup)
+{
+    const auto test_block{CreateTestBlock(test_setup, keys, outputs)};
+    auto pindex{std::make_unique<CBlockIndex>(test_block)};
+    auto test_blockhash{std::make_unique<uint256>(test_block.GetHash())};
+
+    Chainstate& chainstate{test_setup.m_node.chainman->ActiveChainstate()};
+
+    pindex->nHeight = chainstate.m_chain.Height() + 1;
+    pindex->phashBlock = test_blockhash.get();
+    pindex->pprev = chainstate.m_chain.Tip();
+
+    BlockValidationState test_block_state;
+    bench.unit("block").run([&] {
+        LOCK(cs_main);
+        CCoinsViewCache viewNew{&chainstate.CoinsTip()};
+        assert(chainstate.ConnectBlock(test_block, test_block_state, pindex.get(), viewNew));
+    });
+}
+
+static void ConnectBlockAllSchnorr(benchmark::Bench& bench)
+{
+    const std::unique_ptr test_setup{MakeNoLogFileContext<TestChain100Setup>()};
+    auto [keys, outputs]{CreateKeysAndOutputs(test_setup->coinbaseKey, /*num_taproot=*/4, /*num_nontaproot=*/0)};
+    BenchmarkConnectBlock(bench, keys, outputs, *test_setup);
+}
+
+/**
+ * This benchmark is expected to be slower than the AllSchnorr or NoSchnorr benchmark
+ * because it uses transactions with both Schnorr and ECDSA signatures
+ * which requires the transaction to be hashed multiple times for
+ * the different signature allgorithms
+ */
+static void ConnectBlockMixed(benchmark::Bench& bench)
+{
+    const std::unique_ptr test_setup{MakeNoLogFileContext<TestChain100Setup>()};
+    // Blocks in range 848000 to 868000 have a roughly 20 to 80 ratio of schnorr to ecdsa inputs
+    auto [keys, outputs]{CreateKeysAndOutputs(test_setup->coinbaseKey, /*num_taproot=*/1, /*num_nontaproot=*/4)};
+    BenchmarkConnectBlock(bench, keys, outputs, *test_setup);
+}
+
+static void ConnectBlockNoSchnorr(benchmark::Bench& bench)
```

Comment:
And these are the ECDSA ones:
<img width="1000" alt="image" src="https://github.com/user-attachments/assets/46c7b271-1e31-4cad-ba3d-b674acc019bb" />

and

<img width="1000" alt="image" src="https://github.com/user-attachments/assets/eb56e7fe-c76d-477f-830e-a4a679001782" />